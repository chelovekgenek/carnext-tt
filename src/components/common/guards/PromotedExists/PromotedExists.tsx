import React from "react"
import { connect } from "react-redux"
import { Redirect } from "react-router"

import { isPromoted } from "store/entities/promotion"
import { TAppState } from "store/entities"

interface IStateProps {
  promoted: ReturnType<typeof isPromoted>
}
interface IProps extends IStateProps {}

export const PromotedExists: React.FC<IProps> = ({ children, promoted }) => {
  return promoted ? <React.Fragment>{children}</React.Fragment> : <Redirect to="/" />
}

export default connect<IStateProps, {}, {}, TAppState>(state => ({
  promoted: isPromoted(state),
}))(PromotedExists)
