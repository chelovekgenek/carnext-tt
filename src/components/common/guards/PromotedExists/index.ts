export { default as PromotedExists } from "./PromotedExists"
