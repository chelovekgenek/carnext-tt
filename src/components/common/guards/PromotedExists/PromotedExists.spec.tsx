import React from "react"
import { shallow } from "enzyme"

import { PromotedExists } from "./PromotedExists"

describe("Layout", () => {
  const render = (props: React.ComponentProps<typeof PromotedExists>) => shallow(<PromotedExists {...props} />)

  it("should render children if promoted is true", () => {
    expect(render({ promoted: true, children: "children" })).toMatchSnapshot()
  })
  it("should render redirect if promoted is false", () => {
    expect(render({ promoted: false, children: "children" })).toMatchSnapshot()
  })
})
