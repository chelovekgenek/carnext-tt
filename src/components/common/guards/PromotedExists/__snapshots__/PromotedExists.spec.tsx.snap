// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render children if promoted is true 1`] = `
<Fragment>
  children
</Fragment>
`;

exports[`Layout should render redirect if promoted is false 1`] = `
<Redirect
  to="/"
/>
`;
